src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -308,30 +308,17 @@ public async void WorkerRestart(WorkerRestartEvent workerRestart)
 
         private async Task DisposeAndRestartWorkerChannel(string runtime, string workerId)
         {
-            bool isWebHostChannel = await _webHostLanguageWorkerChannelManager.ShutdownChannelIfExistsAsync(runtime, workerId);
-            bool isJobHostChannel = false;
+            _logger.LogDebug("Attempting to dispose webhost or jobhost channel for workerId: {channelId}, runtime:{language}", workerId, runtime);
 
-            // Dispose old worker
-            if (isWebHostChannel)
-            {
-                _logger.LogDebug("Disposing WebHost channel for workerId: {channelId}, for runtime:{language}", workerId, runtime);
-            }
-            else
+            bool isWebHostChannelDisposed = await _webHostLanguageWorkerChannelManager.ShutdownChannelIfExistsAsync(runtime, workerId);
+            bool isJobHostChannelDisposed = await _jobHostLanguageWorkerChannelManager.ShutdownChannelIfExistsAsync(workerId);
```

Comment:
although it's super super small, I think the "TryRemove" in the concurrent dictionary isn't totally trivial: https://stackoverflow.com/questions/15252115/concurrentdictionary-performance-at-a-single-thread-misunderstanding

Maybe worth checking if we need to do "isJobHostChannelDisposed" only if "!isWebHostChannelDisposed" but it's also hard to say without really benchmarking it and it's probably not that important to..!

Reply:
+1 check !isWebHostChannelDisposed before looking at jobHostChannels